components/system-broker/internal/specs/api.go

```diff
@@ -0,0 +1,152 @@
+package specs
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+
+	"github.com/gorilla/mux"
+	"github.com/kyma-incubator/compass/components/director/pkg/graphql"
+	. "github.com/kyma-incubator/compass/components/director/pkg/log"
+	"github.com/kyma-incubator/compass/components/system-broker/internal/director"
+	"github.com/pivotal-cf/brokerapi/v7/domain/apiresponses"
+	"github.com/pkg/errors"
+)
+
+const (
+	SpecsAPI              = "/specifications"
+	AppIDParameter        = "app_id"
+	PackageIDParameter    = "package_id"
+	DefinitionIDParameter = "definition_id"
+)
+
+type SpecsFetcher interface {
+	FindSpecification(ctx context.Context, in *director.PackageSpecificationInput) (*director.PackageSpecificationOutput, error)
+}
+
+func API(rootAPI string, specsFetcher SpecsFetcher) func(router *mux.Router) {
+	handler := &SpecsHandler{
+		specsFetcher: specsFetcher,
+		rootAPI:      rootAPI,
+	}
+	return handler.Routes
+}
+
+type SpecsHandler struct {
+	specsFetcher SpecsFetcher
+	rootAPI      string
+}
+
+func (h *SpecsHandler) Routes(router *mux.Router) {
+	router.HandleFunc(h.rootAPI+SpecsAPI, h.FetchSpec).Methods(http.MethodGet)
+}
+
+func (h *SpecsHandler) FetchSpec(w http.ResponseWriter, req *http.Request) {
```

Comment:
check whether this api behaves the same as the specs api in ORD service and if so, we should probably only have the one there. Referencing it in here (in the OSB metadata of the catalog would require, injecting relevant env vars pointing to ORD svc as well as aligning the security access, which might require some alignment, discussions)

Reply:
Absolutely, maybe it's a good idea to remove the System Broker Specs API in favour of ORD's Specs API but I think that a fair amount of work should be done in regards to this task, so maybe I think that it should be a separate one. Maybe it's out-of-scope for this productisation stage. What do you think?

Reply:
Personally, either is fine by me. @dpanayotov ?

Reply:
out of scope here, will be addressed in another BLI